src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -234,6 +234,11 @@ private async Task ExecuteRulesAsync(IEnumerable<AsyncConsensusRule> asyncRules,
             {
                 ruleContext.ValidationContext.Error = ex.ConsensusError;
             }
+            catch (Exception exception)
+            {
+                this.logger.LogCritical("Not handled error in the rules engine, exception: {0}.", exception.ToString());
```

Comment:
"Unhandled exception in Consensus rules engine : {0}", maybe  ?

Reply:
yes better